samples/attachments-fileshare/AttachmentsFileShare_1/Endpoint/MyHandler.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+
+#region read
+
+public class MyHandler :
+    IHandleMessages<MyMessage>
+{
+    static ILog log = LogManager.GetLogger<MyHandler>();
+
+    public Task Handle(MyMessage message, IMessageHandlerContext context)
+    {
+        log.Info("MyMessage received. Attachment will be streamed to the console.");
+        var attachments = context.Attachments();
+        return attachments.ProcessStream(
```

Comment:
In order handlers you await instead of return, why not do the await approach like in all the others?

Reply:
that is because the delegate inside the ProcessStream does the actual logic. in the other examples a person would await and use the result.